pulser-core/pulser/sampler/samples.py

```diff
@@ -46,21 +45,15 @@ def new_qdict() -> dict:
         }
 
 
-def _pairwise(iterable: typing.Iterable) -> typing.Iterable:
-    a, b = itertools.tee(iterable)
-    next(b, None)
-    return zip(a, b)
-
-
 def _default_to_regular(d: dict | defaultdict) -> dict:
     """Helper function to convert defaultdicts to regular dicts."""
-    if isinstance(d, defaultdict):
+    if isinstance(d, dict):
```

Comment:
So it now catches both defaultdicts and dicts?

Reply:
Yes, it has to so that it can catch defaultdicts inside dicts.

Reply:
Good catch